test/cache_key_test.rb

```diff
@@ -0,0 +1,82 @@
+require 'test_helper'
+
+module IdentityCache
+  class CacheKeyTest < IdentityCache::TestCase
+    def test_cache_key
+      assert_equal(
+        "IDC:7:blob:AssociatedRecord:4733125215075107876:1",
```

Comment:
I think it would be clearer to test the string before hashing rather than after hashing.  That way it will be clearer what has changed in the test code.  It looks like you can do that by testing `CacheKeyGeneration.denormalized_schema_hash`.

If you want to test rails_cache_key, then you can do that using `denormalized_schema_hash` on the expected value (e.g. `"IDC:7:blob:AssociatedRecord:#{CacheKeyGeneration.denormalized_schema_hash(klass)}"`) to simply test `rails_cache_key` without testing the different code paths in `denormalized_schema_hash`.

Reply:
I've added `IdentityCache::CacheKeyGeneration.denormalized_schema_string` to expose the unhashed values and tested against that.